### PR TITLE
STORM-1308

### PR DESCRIPTION
**Pull request by kamleshbhatt**
Please Review.

### CODE REVIEW

**Comment by HeartSaVioR** [comment]
@kamleshbhatt Please remove tick-tuple-test.clj. We don't need both of them.

File: storm-core/test/jvm/org/apache/storm/TickTupleTest.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Map;
+
+public class TickTupleTest {
+
+    @Test
+    public void testTickTupleWorksWithSystemBolt() throws Exception {
+        ILocalCluster cluster = new LocalCluster.Builder().withSimulatedTime().withNimbusDaemon(true).build();
```
Comment (HeartSaVioR) [inline storm-core/test/jvm/org/apache/storm/TickTupleTest.java]: We can apply try-finally to ensure cluster will be closed when end of test.

File: storm-core/test/jvm/org/apache/storm/TickTupleTest.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            public void declareOutputFields(OutputFieldsDeclarer ofd) {}
+
+            private void writeObject(java.io.ObjectOutputStream stream) {}
+        };
+    }
+
+    private StormTopology createNoOpTopology() {
+        TopologyBuilder builder = new TopologyBuilder();
+        builder.setSpout("1", makeNoOpSpout());
+        builder.setBolt("2", makeNoOpBolt()).shuffleGrouping("1");
```
Comment (HeartSaVioR) [inline storm-core/test/jvm/org/apache/storm/TickTupleTest.java]: I guess it was field grouping, not shuffle grouping.

```
(Thrift/buildTopology
                    {"1" (Thrift/prepareSpoutDetails noop-spout)}
                    {"2" (Thrift/prepareBoltDetails
                           {(Utils/getGlobalStreamId "1" nil)
                            (Thrift/prepareFieldsGrouping ["tuple"])}
                           noop-bolt)})
```

**Comment by HeartSaVioR** [comment]
Left some comments but otherwise looks good.

**Comment by HeartSaVioR** [comment]
@kamleshbhatt Could you let me know your Apache JIRA ID so that I can assign this issue to you?

**Comment by kamleshbhatt** [comment]
Thanks for reviewing the patch @HeartSaVioR  .
Modified the pull request as per review comments.  Please take a look.
My jira id is "kbhatt".

**Comment by kamleshbhatt** [comment]
@HeartSaVioR Could you please review the revised pull request.

File: storm-core/test/jvm/org/apache/storm/MessagingTest.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+import org.apache.storm.generated.StormTopology;
+import org.apache.storm.testing.*;
+import org.apache.storm.topology.TopologyBuilder;
+import org.apache.storm.utils.Utils;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.*;
+
+/**
```
Comment (HeartSaVioR) [inline storm-core/test/jvm/org/apache/storm/MessagingTest.java]: Please remove this javadoc comments.

Comment (HeartSaVioR) [inline storm-core/test/jvm/org/apache/storm/MessagingTest.java]: Oh I found this is removed after. It's OK.

**Comment by HeartSaVioR** [comment]
+1 build failure are not relevant (integration test failure)

**Pull request closed without merging**